service_layer_test.cc

```diff
@@ -280,6 +282,28 @@ TEST(ServiceLayerFunctionalityTest, StreamInvalid) {
   delete chirp1;
 }
 
+// tests start_stream with invalid hashtags
+// start_stream should return false for an invalid hashtag
+TEST(ServiceLayerFunctionalityTest, InvalidHashtag) {
+  ServiceLayerFunctionality sl_func(true);
+  sl_func.registeruser("jillian");
+  EXPECT_EQ(sl_func.user_exists("jillian"), true);
+
+  // start_stream should not accept a hashtag without characters after it
+  bool start = sl_func.start_stream("jillian", "#");
+  EXPECT_EQ(start, false);
+
+  // start_stream should not accept a hashtag without a # in front
+  start = sl_func.start_stream("jillian", "athenahacks");
+  EXPECT_EQ(start, false);
+  start = sl_func.start_stream("jillian", "athena#hacks");
+  EXPECT_EQ(start, false);
+
```

Comment:
What about the case where you have "#athena#hacks"? Would the hashtag be registered as "athena" or "athena#hacks"?

Reply:
oh shoot you're right, I have no idea. I think under my implementation it would be registered as athena#hacks since the hashtag must end with a space or the end of the string but maybe @barath can clarify what the program behavior should be